week_1/String/find_all_anagrams/find_all_anagrams.js

```diff
@@ -0,0 +1,67 @@
+// Given two strings long and short, return an array of all the start indices of short's anagrams in long. You may return the answer in any order.
+/**
+ * @param {string} long A longer string
+ * @param {string} short A shorter string
+ * @return {number[]}
+ */
+
+const findAllAnagrams = (long, short) => {
+	if (long.length < short.length) return [];
+
+	const indices = [];
+	// Create a Map that will contain all the values of the shorter string
+	const shortMap = new Map();
+	for (char of short) {
+		shortMap.set(char, shortMap.get(char) + 1 || 1);
```

Comment:
I don't love potentially adding `undefined` with `1`. This line wouldn't be valid in most type safe languages, including TypeScript. We should consider breaking this line in a few lines to avoid potentially adding `undefined`